File: Misc/FizzBuzz/TS/fizzBuzz.ts
function fizzBuzz(n: number): string[]{
    let results = [];

    for(let i = 1; i <= n; i++){
        const divisibleBy3 = i % 3 === 0;
        const divisibleBy5 = i % 5 === 0;

        if(divisibleBy3 && divisibleBy5){
            results.push("FizzBuzz")
        } else if(divisibleBy3){
            results.push("Fizz")
        } else if(divisibleBy5) {
            results.push("Buzz")
        } else{
            results.push(i.toString())
        }
    }

    return results
}

const testCases: number[] = [1, 3, 5, 15, 16];

for (const n of testCases) {
    console.log(`Input:       ${n}`);
    console.log(`Output:      ${JSON.stringify(fizzBuzz(n))}`);
    console.log("-".repeat(35));
}

File: Misc/FizzBuzz/python/fizzbuzz.py
class Solution:
    @staticmethod
    def fizzbuzz(n: int) -> list[int]:
        results = []
        for i in range (1, n+1):
            divisible_by_3 = i % 3 == 0
            divisible_by_5 = i % 5 == 0
            
            if (divisible_by_3 and divisible_by_5):
                results.append("FizzBuzz")
            elif (divisible_by_3):
                results.append("Fizz")
            elif (divisible_by_5):
                results.append("Buzz")
            else:
                results.append(str(i))
        return results
    
if __name__ == "__main__":
    test_cases = [1, 3, 5, 15, 16]

    for n in test_cases:
        print(f"Input:       {n}")
        print(f"Output:      {Solution.fizzbuzz(n)}")
        print("-" * 35)
    